fenix/app/src/main/java/org/mozilla/fenix/translations/preferences/downloadlanguages/DownloadLanguageFileDialogPreference.kt

```diff
@@ -0,0 +1,184 @@
+                    fileSize.toMegabyteOrKilobyteString(),
+                ),
+                modifier = Modifier
+                    .semantics { heading() }
+                    .padding(start = 16.dp, end = 16.dp),
+                color = FirefoxTheme.colors.textPrimary,
+                style = FirefoxTheme.typography.headline7,
+                fontWeight = FontWeight.Bold,
+            )
+
+            if (isAllLanguagesItemType) {
+                Text(
+                    text = stringResource(
+                        R.string.download_language_file_dialog_message_all_languages,
+                    ),
+                    modifier = Modifier.padding(16.dp),
+                    style = FirefoxTheme.typography.body2,
+                    color = FirefoxTheme.colors.textPrimary,
+                )
+            }
+
+            DownloadLanguageFileDialogCheckbox(
+                isCheckBoxEnabled = isCheckBoxEnabled,
+                onSavingModeStateChange = onSavingModeStateChange,
+            )
+
+            val primaryButtonText: String = if (isAllLanguagesItemType) {
+                stringResource(id = R.string.download_language_file_dialog_positive_button_text_all_languages)
+            } else {
+                stringResource(id = R.string.download_language_file_dialog_positive_button_text)
+            }
+
+            PrimaryButton(
+                text = primaryButtonText,
+                modifier = Modifier
+                    .padding(start = 16.dp, end = 16.dp, top = 16.dp)
+                    .fillMaxWidth(),
+                onClick = {
+                    onConfirmDownload()
+                },
+            )
+
+            TextButton(
+                text = stringResource(id = R.string.download_language_file_dialog_negative_button_text),
+                modifier = Modifier
+                    .padding(start = 16.dp, end = 16.dp)
+                    .fillMaxWidth(),
+                onClick = {
+                    onCancel()
+                },
+            )
+        }
+    }
+}
+
+@Composable
+private fun DownloadLanguageFileDialogCheckbox(
+    isCheckBoxEnabled: Boolean,
+    onSavingModeStateChange: (Boolean) -> Unit,
+) {
+    Row(
+        modifier = Modifier
+            .defaultMinSize(minHeight = 56.dp)
+            .padding(start = 10.dp, end = 16.dp),
+    ) {
+        Checkbox(
+            modifier = Modifier.align(Alignment.CenterVertically),
+            checked = isCheckBoxEnabled,
+            onCheckedChange = onSavingModeStateChange,
+            colors = CheckboxDefaults.colors(
+                checkedColor = FirefoxTheme.colors.formSelected,
+                uncheckedColor = FirefoxTheme.colors.formDefault,
+            ),
+        )
+
+        Spacer(modifier = Modifier.width(20.dp))
+
+        Text(
+            modifier = Modifier
+                .align(Alignment.CenterVertically),
+            text = stringResource(
```

Comment:
Could you take a look at the checkbox again? It needs a slight adjustment so the checkbox and label are programmatically connected in TalkBack.

In TalkBack, the checkbox is not programmatically connected to the label of "Always download in data saving mode" and shows up as a freestanding element.

Reply:
Done

Reply:
Thanks for the update! They are connected now, but the checkbox is not toggable in TalkBack. Could you take one more look at it?

It is correctly reporting the name, role, value, but the checkbox doesn't allow me to toggle it.

Reply:
Done , I fixed it.

Reply:
Perfect, thanks!